projects/igniteui-angular/src/lib/grids/column.component.ts

```diff
@@ -1032,6 +1032,10 @@ export class IgxColumnComponent implements AfterContentInit {
         this._pinned = false;
 
         const targetColumn = grid._unpinnedColumns[index];
+
+        const args = { column: this, insertAtIndex: -1, columnIsPinned: false };
+        grid.onColumnPinning.emit(args);
```

Comment:
I am not sure `insertAtIndex` should remain -1 after this change. By itself it does not carry the notion that it is **the index the column is pinned at**. If you introduce a property for whether the column is being pinned or not, then this property is the new index in its respective collection, i.e. the index the column will have in the **unpinned** collection.